cmd/mesh/profile-dump.go

```diff
@@ -0,0 +1,161 @@
+	cmd.PersistentFlags().BoolVarP(&args.helmValues, "helm-values", "", false,
+		"If set, dumps the Helm values that IstioControlPlaceSpec is translated to before manifests are rendered.")
+	cmd.PersistentFlags().StringVarP(&args.set, "set", "s", "", setFlagHelpStr)
+}
+
+func profileDumpCmd(rootArgs *rootArgs, pdArgs *profileDumpArgs) *cobra.Command {
+	return &cobra.Command{
+		Use:   "dump",
+		Short: "Dumps an Istio configuration profile.",
+		Long:  "The dump subcommand is used to dump the values in an Istio configuration profile.",
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, args []string) {
+			profileDump(rootArgs, pdArgs)
+		}}
+
+}
+
+func profileDump(args *rootArgs, pdArgs *profileDumpArgs) {
+	checkLogsOrExit(args)
+
+	writer, err := getWriter("")
+	if err != nil {
+		logAndFatalf(args, err.Error())
+	}
+	defer func() {
+		if err := writer.Close(); err != nil {
+			logAndFatalf(args, "Did not close output successfully: %v", err)
+		}
+	}()
+
+	mergedYAML := ""
+	mergedcps := &v1alpha2.IstioControlPlaneSpec{}
+
+	// TODO(mostrowski): load profile if set using --set flag.
+	overlayYAML := ""
+	if pdArgs.inFilename != "" {
+		b, err := ioutil.ReadFile(pdArgs.inFilename)
+		if err != nil {
+			logAndFatalf(args, "Could not read values file %f: %s", pdArgs.inFilename, err)
+		}
+		overlayYAML = string(b)
+	}
+
+	// Start with unmarshaling and validating the user CR (which is an overlay on the base profile).
+	overlayICPS := &v1alpha2.IstioControlPlaneSpec{}
+	if err := util.UnmarshalWithJSONPB(overlayYAML, overlayICPS); err != nil {
+		logAndFatalf(args, "Could not unmarshal the input file: %s\n\nOriginal YAML:\n%s\n", err, overlayYAML)
+	}
+	if errs := validate.CheckIstioControlPlaneSpec(overlayICPS, false); len(errs) != 0 {
+		logAndFatalf(args, "Input file failed validation with the following errors: %s\n\nOriginal YAML:\n%s\n", errs, overlayYAML)
+	}
+
+	// Now read the base profile specified in the user spec.
+	fname, err := helm.FilenameFromProfile(overlayICPS.Profile)
+	if err != nil {
+		logAndFatalf(args, "Could not get filename from profile: %s", err)
+	}
+
+	baseYAML, err := helm.ReadValuesYAML(overlayICPS.Profile)
+	if err != nil {
+		logAndFatalf(args, "Could not read the profile values for %s: %s", fname, err)
+	}
+
+	mergedYAML, err = helm.OverlayYAML(baseYAML, overlayYAML)
+	if err != nil {
+		logAndFatalf(args, "Could not overlay user config over base: %s", err)
+	}
+	// Now unmarshal and validate the combined base profile and user CR overlay.
+	if err := util.UnmarshalWithJSONPB(mergedYAML, mergedcps); err != nil {
+		logAndFatalf(args, err.Error())
+	}
+	if errs := validate.CheckIstioControlPlaneSpec(mergedcps, true); len(errs) != 0 {
+		logAndFatalf(args, err.Error())
+	}
+
+	t, err := translate.NewTranslator(version.NewMinorVersion(1, 2))
+	if err != nil {
+		logAndFatalf(args, "%s", err)
+	}
+
+	if pdArgs.helmValues {
+		mergedYAML, err = component.CompileHelmValues(mergedcps, t, "")
```

Comment:
although this one was merged already, but I think would it be better to rename that as TranslateToHelmValues

Reply:
Done.